### DISCUSSION TITLE
How to convert vector to python tuple?

### DISCUSSION THREAD

**Discussion by cgarciae** [wjakob/nanobind#838] [Q&A]
I need to convert a variable length std::vector to a python tuple. What is the most efficient way to do this?

**Comment by wjakob** [comment]
The builtin type caster will generate a list. The most efficient way is to write this yourself using Python C API. 
```cpp
        nb::object r = nb::steal(PyTuple_New(size));
        for (Py_ssize_t i = 0; i < size; ++i)
            NB_TUPLE_SET_ITEM(r.ptr(), i, nb::cast(...).release().ptr());
```
